vividus-plugin-web-app-playwright/src/main/java/org/vividus/ui/web/playwright/steps/WaitSteps.java

```diff
@@ -0,0 +1,109 @@
+import org.vividus.ui.web.playwright.BrowserContextProvider;
+import org.vividus.ui.web.playwright.UiContext;
+import org.vividus.ui.web.playwright.locator.PlaywrightLocator;
+
+public class WaitSteps
+{
+    private final UiContext uiContext;
+    private final ISoftAssert softAssert;
+    private final BrowserContextProvider browserContextProvider;
+
+    public WaitSteps(UiContext uiContext, ISoftAssert softAssert, BrowserContextProvider browserContextProvider)
+    {
+        this.uiContext = uiContext;
+        this.softAssert = softAssert;
+        this.browserContextProvider = browserContextProvider;
+    }
+
+    /**
+     * Waits for appearance of an <b><i>element</i></b> with the specified <b>locator</b>
+     * @param locator locator to locate element
+     */
+    @When("I wait until element located by `$locator` appears")
+    public void waitForElementAppearance(PlaywrightLocator locator)
+    {
+        waitForElementState(locator, WaitForSelectorState.VISIBLE);
+    }
+
+    /**
+     * Waits for disappearance of an <b><i>element</i></b> with the specified <b>locator</b>
+     * @param locator locator to locate element
+     */
+    @When("I wait until element located by `$locator` disappears")
+    public void waitForElementDisappearance(PlaywrightLocator locator)
+    {
+        waitForElementState(locator, WaitForSelectorState.HIDDEN);
+    }
+
+    /**
+     * Waits for expected number of elements.
+     * @param locator        The locator to find elements.
+     * @param comparisonRule The rule to match the variable value. The supported rules:
+     *                       <ul>
+     *                       <li>less than (&lt;)</li>
+     *                       <li>less than or equal to (&lt;=)</li>
+     *                       <li>greater than (&gt;)</li>
+     *                       <li>greater than or equal to (&gt;=)</li>
+     *                       <li>equal to (=)</li>
+     *                       <li>not equal to (!=)</li>
+     *                       </ul>
+     * @param number         The expected number of elements.
+     */
+    @When("I wait until number of elements located by `$locator` is $comparisonRule $number")
+    public void waitForElementNumber(PlaywrightLocator locator, ComparisonRule comparisonRule, int number)
+    {
+        BooleanSupplier waitCondition = () -> comparisonRule.getComparisonRule(number)
+                .matches(uiContext.locateElement(locator).count());
+        String assertionMessage = String.format(
+                "Failed to wait for number of elements located by '%s' to be %s %d", locator, comparisonRule,
+                number);
+        runWithTimeoutAssertion(assertionMessage, () -> browserContextProvider.get().waitForCondition(waitCondition));
+    }
+
+    private void waitForElementState(PlaywrightLocator locator, WaitForSelectorState state)
+    {
+        Locator.WaitForOptions waitOption = new Locator.WaitForOptions().setState(state);
```

Comment:
is it possible to configure wait timeout and polling duration? (in the similar way like  we have `ui.wait.timeout` and `ui.wait.polling-period}`)

Reply:
It's only possible to configure the wait timeout but not the polling interval.